src/rt/cpp/cown.h

```diff
@@ -120,19 +121,30 @@ namespace verona::cpp
      */
     cown_ptr& operator=(cown_ptr&& other)
     {
+      clear();
       allocated_cown = other.allocated_cown;
       other.allocated_cown = nullptr;
       return *this;
     }
 
-    ~cown_ptr()
+    /**
+     * Sets the cown_ptr to nullptr, and decrements the reference count
+     * if it was not already nullptr.
+     */
+    void clear()
     {
       // Condition to handle moved cown ptrs.
       if (allocated_cown != nullptr)
       {
         auto& alloc = verona::rt::ThreadAlloc::get();
         verona::rt::Cown::release(alloc, allocated_cown);
       }
+      allocated_cown = nullptr;

```

Comment:
Nitpick, but this could just go in the conditional right? Although it is more readable this way and maybe I am unaware of some codegen benefits.